pkg/tcpip/network/ipv4/icmp.go

```diff
@@ -134,6 +134,44 @@ func (e *endpoint) handleICMP(r *stack.Route, netHeader buffer.View, vv buffer.V
 	case header.ICMPv4Timestamp:
 		received.Timestamp.Increment()
 
+		// Only send a reply if the checksum is valid.
+		wantChecksum := h.Checksum()
+		// Reset the checksum field to 0 to can calculate the proper
+		// checksum. We'll have to reset this before we hand the packet off.
+		h.SetChecksum(0)
+
+		// It's possible that a raw socket expects to receive this.
+		h.SetChecksum(wantChecksum)
+		e.dispatcher.DeliverTransportPacket(r, header.ICMPv4ProtocolNumber, netHeader, vv)
+
+		gotChecksum := ^header.ChecksumVV(vv, 0 /* initial */)
+		if gotChecksum != wantChecksum {
+			received.Invalid.Increment()
+			return
+		}
+
+		vv := vv.Clone(nil)
+		vv.TrimFront(header.ICMPv4TimeStampMinimumSize)
+		hdr := buffer.NewPrependable(int(r.MaxHeaderLength()) + header.ICMPv4TimeStampMinimumSize)
+		pkt := header.ICMPv4(hdr.Prepend(header.ICMPv4TimeStampMinimumSize))
+		copy(pkt, h)
+		pkt.SetType(header.ICMPv4TimestampReply)
+
+		// millisecondsPerDay is Milli Seconds in a day calculation for Timestamp reply messages.
+		lastDay := now.Truncate(24 * time.Hour)
```

Comment:
now does not seem to be defined? Also I believe as per RFC this needs to be UTC.